sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -393,6 +393,60 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * SPARK-28443: fail the DDL command if it creates a table with null-type columns
+ */
+object DDLCheck extends (LogicalPlan => Unit) {
+
+  def failAnalysis(msg: String): Unit = { throw new AnalysisException(msg) }
+
+  def throwWhenExistsNullType(dataType: DataType): Unit = {
+    dataType match {
+      case ArrayType(elementType, _) =>
+        throwWhenExistsNullType(elementType)
+
+      case MapType(keyType, valueType, _) =>
+        throwWhenExistsNullType(keyType)
+        throwWhenExistsNullType(valueType)
+
+      case StructType(fields) =>
+        fields.foreach{field => throwWhenExistsNullType(field.dataType)}
+
+      case other if other == NullType =>
+        failAnalysis("DataType NullType is not supported for create table")
+
+      case _ => // OK
+    }
+  }
+
+  def checkSchema(schema: StructType): Unit = {
+    schema.foreach{field => throwWhenExistsNullType(field.dataType)}
+  }
+
+  override def apply(plan: LogicalPlan): Unit = {
+    plan.foreach {
+      case CreateTable(tableDesc, _, _) =>
+        checkSchema(tableDesc.schema)
+
+      case CreateV2Table(_, _, tableSchema, _, _, _) =>
```

Comment:
I think the rationale here is that, we don't want to create tables with null type field. It should be same for both `CREATE TABLE` and `CTAS`.

Reply:
OK, and also add `AlterTable`.